www/%username/settings/edit.spt

```diff
@@ -25,8 +28,34 @@ if 'new-password' in body:
         )
         if not p:
             raise response.error(403, _("Incorrect password"))
+
+    if body.get('confirmed') != 'true':
+        passhash = sha1(body['new-password'].encode("utf-8")).hexdigest().upper()
+        passhash_short = passhash[:5]
+        URL = "https://api.pwnedpasswords.com/range/"+passhash_short
+
+        r = requests.get(url = URL)
+        count = 0
+        for line in r.text.split("\n"):
+            suffix = line.split(":")[0]
+            if passhash_short + suffix == passhash:
+                count = int(line.split(":")[1].strip())
+        strength, improvements = passwordmeter.test(body['new-password'])
+
+        back_to = p.path('settings')
+        if strength > 0.3 and count < 2000 :
+            msg = _(
+                "Are you sure you want to change your password?"
+            )
+            raise response.render('templates/confirm.spt', state, cls='warning', msg=msg, back_to=back_to)
```

Comment:
This confirmation warning is unnecessary, we should only warn the user when the password is weak or compromised.